### PR TITLE
io.open → open

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
In Python 3, `io.open()` is an alias for the builtin `open()` function:
https://docs.python.org/3/library/io.html#io.open

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [x] Changes documented in docs/release.rst
* [x] GitHub Actions have all passed
* [x] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by DimitriPapadopoulos** [comment]
That's disturbing. Despite `io.open()` being (supposedly) an alias for `open()`, changing to `open()` fails. The error message is unclear, that's perhaps a real issue.

**Comment by DimitriPapadopoulos** [comment]
No unexpected failures locally 🤯

Any clue?

**Comment by DimitriPapadopoulos** [comment]
This should be fixed by changing the name of the local function `open`, for example `zarr_open`.

Then change:
https://github.com/zarr-developers/zarr-python/blob/4132f360616a4c8bfa3dd4e979a4793c5d84cdfc/zarr/__init__.py#L3-L5
to:
```python
from zarr.convenience import (consolidate_metadata, copy, copy_all, copy_store,
                              load, open_consolidated, save, save_array,
                              save_group, tree)
from zarr.convenience import zarr_open as open
```

**Comment by d-v-b** [comment]
@DimitriPapadopoulos 
> This should probably be fixed by changing the name of the local function open, perhaps to zarr_open.

I have no objection to this strategy. Do you want to pursue that in this PR, or elsewhere?

File: zarr/convenience.py [hunk 1]
```diff
@@ -491,7 +492,7 @@ def __init__(self, log):
         elif callable(log):
             self.log_func = log
         elif isinstance(log, str):
-            self.log_file = io.open(log, mode="w")
+            self.log_file = _builtin_open(log, mode="w")
```
Comment (rabernat) [inline zarr/convenience.py:495]: Are we certain this is the only use of `io.open`?

Comment (DimitriPapadopoulos) [reply zarr/convenience.py:495]: I think so. A grep of `io.open` yields:
```
$ fgrep -R io.open --exclude-dir=.git
zarr/convenience.py:            self.log_file = io.open(log, mode="w")
build/lib/zarr/convenience.py:            self.log_file = io.open(log, mode='w')
$ 
```
Do you have a specific concern, anything I may have overlooked?

**Review by rabernat — APPROVED** [review]
Thanks a lot for this PR!

File: zarr/convenience.py (hunk 1 above)
Comment (rabernat) [reply zarr/convenience.py:495]: No not really. Just wanted to make sure we only have to do this once.

**Pull request merged**